### PR TITLE
Add support for ignoring delegated properties

### PR DESCRIPTION
**Pull request by clementetb**
Delegated properties did not support the `@Ignore` annotation. This was preventing the users to write their own custom delegated properties.

This would help users to write delegates on Realm types to do any conversion to their own custom types.

#closes https://github.com/realm/realm-kotlin/issues/1381

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,31 @@
+## 1.8.1-SNAPSHOT (YYYY-MM-DD)
+
+### Breaking Changes
+* None.
+
+### Enhancements
+* Realm objects now support ignoring delegated properties . (Issue [#1377](https://github.com/realm/realm-kotlin/pull/1386))
```
Comment (cmelchior) [inline CHANGELOG.md:7]:
```suggestion
* Realm objects now support ignoring delegated properties. (Issue [#1377](https://github.com/realm/realm-kotlin/pull/1386))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,31 @@
+## 1.8.1-SNAPSHOT (YYYY-MM-DD)
```
Comment (cmelchior) [inline CHANGELOG.md:1]: This should be 1.9.0

**Review by cmelchior — APPROVED** [review]
:shipit:

File: CHANGELOG.md (hunk 2 above)
Comment (clementetb) [reply CHANGELOG.md:1]: It is ok, this PR should target releases.

**Pull request merged**